llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -9425,6 +9425,21 @@ isImpliedCondICmps(CmpPredicate LPred, const Value *L0, const Value *L1,
       return true;
   }
 
+  // a - b == NonZero -> a != b
+  // ptrtoint(a) - ptrtoint(b) == NonZero -> a != b
+  const APInt *RHSC;
```

Comment:
minor nit think that the name RHSC is a bit confusing when there is RHS that is a icmp already. maybe L1C